src/game/models/mod.rs

```diff
@@ -105,6 +144,15 @@ impl From<Size> for Vector {
     }
 }
 
+impl From<[f64; 2]> for Vector {
```

Comment:
Might be a good idea. It's a shame that the rust standard library doesn't implement Add for list types. Even if it couldn't be done in the worst case, it could be hard-coded in specific small-array cases like they did with pretty much every trait on here: https://doc.rust-lang.org/std/primitive.array.html